app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/VatDetailsController.scala

```diff
@@ -81,8 +77,14 @@ class VatDetailsController @Inject() (
     hc: HeaderCarrier,
     request: Request[AnyContent]
   ): Future[Result] =
-    vatControlListConnector.vatControlList(VatControlListRequest(vatForm.number)).flatMap {
-      case Right(vatControlListResponse) =>
+    vatControlListConnector.vatControlList(VatControlListRequest(vatForm.number)).foldF(
+      responseError =>
+        responseError.status match {
+          case NOT_FOUND | BAD_REQUEST =>
+            Future.successful(Redirect(VatDetailsController.vatDetailsNotMatched(service)))
+          case SERVICE_UNAVAILABLE => Future.successful(Results.ServiceUnavailable(errorTemplate(service)))
```

Comment:
This isn't comprehensive, right? We've only checked three possible integers here for the status. What happens when we get a 500 for example?

Reply:
I have added `case _ => Future.successful(Results.InternalServerError(errorTemplate(service)))` to handle other unexpected scenarios - happy to discuss if you like